### PR TITLE
redis-plus-plus: conan v2 support + fix `build_async=True`

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/13876

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/redis-plus-plus/all/conanfile.py [hunk 1]
```diff
@@ -11,129 +16,137 @@ class RedisPlusPlusConan(ConanFile):
...
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support.",
+            )
+
+        if self.info.options.with_tls and not self.dependencies["hiredis"].options.with_ssl:
+            raise ConanInvalidConfiguration(f"{self.name}:with_tls=True requires hiredis:with_ssl=True")
```
Comment (prince-chrismc) [inline recipes/redis-plus-plus/all/conanfile.py:85]:
```suggestion
            raise ConanInvalidConfiguration(f"{self.ref}:with_tls=True requires hiredis:with_ssl=True")
```

**Review by prince-chrismc — APPROVED** [review]
Just a nice for 2.0 print out

File: recipes/redis-plus-plus/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/redis-plus-plus/all/conanfile.py:85]: it's not a valid pattern for command line or profile so I don't recommend this modification

Comment (prince-chrismc) [reply recipes/redis-plus-plus/all/conanfile.py:85]: In 2.0 you need to pass the version? 🤔 not important

**Review by toge — APPROVED** [review]

**Pull request merged**